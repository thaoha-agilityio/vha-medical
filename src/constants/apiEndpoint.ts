export const API_ENDPOINT = {
  AUTH: '/auth/local',
  APPOINTMENTS: '/appointments',
  NOTIFICATIONS: '/notifications',
  SPECIALTIES: '/specialties',
  USERS: '/users',
  UPLOAD: '/upload',
  CHEMISTS: '/chemists',
  PERMISSIONS: '/users-permissions',
};

export const API_URL = `${process.env.NEXT_PUBLIC_API_URL}/api`;
export const IMGBB_URL = `${process.env.NEXT_PUBLIC_IMGBB_URL}?key=${process.env.NEXT_PUBLIC_IMGBB_API_KEY}`;

export const AUTH_TOKEN = process.env.NEXT_PUBLIC_AUTH_TOKEN;
