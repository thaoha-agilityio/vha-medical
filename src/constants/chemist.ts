import { UserModel } from '@/types';

export const DEFAULT_CHEMIST_DATA: UserModel = {
  email: '',
  username: '',
  password: '',
  description: '',
  rating: 0,
  tasks: 0,
  reviews: 0,
  specialtyId: undefined,
  avatar: undefined,
};

export const CHEMISTS_SEARCH_PARAMS = ['specialtyId'] as const;
