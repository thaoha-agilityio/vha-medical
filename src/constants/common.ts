export const RESULT_NOT_FOUND = 'Result Not Found';
