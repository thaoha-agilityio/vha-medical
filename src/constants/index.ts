export * from './apiEndpoint';
export * from './appointment';
export * from './common';
export * from './firebase';
export * from './image';
export * from './message';
export * from './navigation';
export * from './pagination';
export * from './regex';
export * from './router';
export * from './timing';
export * from './chemist';
export * from './encode';
