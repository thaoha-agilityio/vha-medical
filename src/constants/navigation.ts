import {
  AppointmentIcon,
  ChemistIcon,
  DashboardIcon,
  DoctorIcon,
  HospitalIcon,
  MedicineIcon,
  MessageIcon,
  SettingIcon,
} from '@/icons';

// Router
import { PRIVATE_ROUTES } from './router';

export const NAVBAR_LINKS = [
  {
    name: 'Dashboard',
    href: PRIVATE_ROUTES.DASHBOARD,
    icon: DashboardIcon,
    isEnable: true,
  },
  {
    name: 'Appointments',
    href: PRIVATE_ROUTES.APPOINTMENTS,
    icon: AppointmentIcon,
    isEnable: true,
  },
  {
    name: 'Chemists',
    href: PRIVATE_ROUTES.CHEMISTS,
    icon: ChemistIcon,
    isEnable: true,
  },
  { name: 'Hospitals', href: '#', icon: HospitalIcon },
  { name: 'Doctors', href: '#', icon: DoctorIcon },
  { name: 'Medicines', href: '#', icon: MedicineIcon },
];

export const SUPPORT_LINKS = [
  { name: 'Messages', href: '#', icon: MessageIcon },
  {
    name: 'Settings',
    href: '#',
    icon: SettingIcon,
  },
];
