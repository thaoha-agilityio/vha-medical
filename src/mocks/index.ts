export * from './appointments';
export * from './auth';
export * from './chemists';
export * from './common';
export * from './dateTime';
export * from './notification';
export * from './options';
export * from './profile';
export * from './table';
export * from './user';
