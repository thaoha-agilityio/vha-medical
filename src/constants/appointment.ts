import { AppointmentModel, AppointmentStatusOption } from '@/types';
import { DEFAULT_CHEMIST_DATA } from './chemist';

export const APPOINTMENT_STATUS_OPTIONS: AppointmentStatusOption[] = [
  {
    key: 'new',
    label: 'New',
    value: 0,
  },
  {
    key: 'meeting',
    label: 'Meeting',
    value: 1,
  },
  {
    key: 'cancelled',
    label: 'Cancelled',
    value: 2,
  },
];

export const APPOINTMENT_STATUS = [
  {
    key: '0',
    label: 'New',
  },
  {
    key: '1',
    label: 'Meeting',
  },
  {
    key: '2',
    label: 'Canceled',
  },
];

export const DEFAULT_APPOINTMENT_DATA: AppointmentModel = {
  startTime: '',
  durationTime: '',
  status: 0,
  receiverId: {
    data: {
      id: '',
      attributes: DEFAULT_CHEMIST_DATA,
    },
  },
  senderId: {
    data: {
      id: '',
      attributes: DEFAULT_CHEMIST_DATA,
    },
  },
};

export const APPOINTMENT_SEARCH_PARAMS = ['receiverId', 'senderId'] as const;
