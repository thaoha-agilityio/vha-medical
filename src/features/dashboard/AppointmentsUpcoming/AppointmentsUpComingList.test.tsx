import { usePathname, useRouter, useSearchParams } from 'next/navigation';
import { act, fireEvent, render, screen } from '@testing-library/react';

// Constants
import { PRIVATE_ROUTES } from '@/constants';

// Mocks
import { MOCK_APPOINTMENTS, MOCK_USERS_LOGGED } from '@/mocks';

// Components
import AppointmentsUpcomingList, {
  AppointmentsUpcomingListProps,
} from './AppointmentsUpcomingList';

jest.mock('../../../services/appointment.ts', () => ({
  getAppointments: jest.fn(),
}));

jest.mock('next/navigation', () => ({
  usePathname: jest.fn(),
  useSearchParams: jest.fn(),
  useRouter: jest.fn(),
}));

jest.mock('@/utils', () => ({
  ...jest.requireActual('@/utils'),
  cn: jest.fn(() => 'mocked-class'),
}));

describe('AppointmentsUpComingList test cases', () => {
  const mockProps: AppointmentsUpcomingListProps = {
    appointments: MOCK_APPOINTMENTS,
    defaultStatus: 'new',
    pagination: {
      page: 1,
      pageSize: 10,
      pageCount: 1,
      total: 3,
    },
    userLogged: MOCK_USERS_LOGGED[0],
  };
  const mockReplace = jest.fn();
  const mockUsePathname = usePathname as jest.Mock;
  const mockUseRouter = useRouter as jest.Mock;
  const mockUseSearchParams = useSearchParams as jest.Mock;
  const mockSearchParams = new URLSearchParams();

  const setup = async (props: AppointmentsUpcomingListProps) =>
    act(() => render(<AppointmentsUpcomingList {...props} />));

  beforeEach(() => {
    mockUsePathname.mockReturnValue(PRIVATE_ROUTES.DASHBOARD);
    mockUseRouter.mockReturnValue({ replace: mockReplace });
    mockUseSearchParams.mockReturnValue(mockSearchParams);
  });

  afterEach(() => jest.clearAllMocks());

  it('should render correctly', async () => {
    const { asFragment } = await setup(mockProps);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should be able to filter by status', async () => {
    await setup(mockProps);

    const statusSelect = screen.getByRole('button', {
      name: /new appointment status/i,
    });

    fireEvent.click(statusSelect);

    const statusOption = screen.getByRole('option', {
      name: /meeting/i,
    });

    fireEvent.click(statusOption);

    expect(statusSelect.textContent).toBe('Meeting');
    expect(mockReplace).toHaveBeenCalled();
  });
});
