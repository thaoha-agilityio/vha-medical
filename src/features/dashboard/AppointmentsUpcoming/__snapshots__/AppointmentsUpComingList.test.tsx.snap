// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppointmentsUpComingList test cases should render correctly 1`] = `
<DocumentFragment>
  <section
    class="flex flex-col relative overflow-hidden text-foreground box-border outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium rounded-large transition-transform-background motion-reduce:transition-none w-full xl:max-w-[320px] 2xl:max-w-[550px] h-fit py-4 px-5 bg-background-200"
    tabindex="-1"
  >
    <div
      class="flex justify-between items-center"
    >
      <p
        class="mocked-class"
      >
        Appointments
      </p>
      <div>
        <div
          class="group flex flex-col w-full transition-background motion-reduce:transition-none !duration-150 group relative justify-end data-[has-label=true]:mt-[calc(theme(fontSize.small)_+_10px)] max-w-[102px] max-h-[36px]"
          data-filled="true"
          data-has-value="true"
          data-slot="base"
        >
          <div
            aria-hidden="true"
            data-a11y-ignore="aria-hidden-focus"
            data-testid="hidden-select-container"
            style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
          >
            <input
              style="font-size: 16px;"
              tabindex="0"
              type="text"
              value="new"
            />
            <label>
              <select
                size="3"
                tabindex="-1"
                title=""
              >
                <option />
                <option
                  value="new"
                >
                  New
                </option>
                <option
                  value="meeting"
                >
                  Meeting
                </option>
                <option
                  value="cancelled"
                >
                  Cancelled
                </option>
              </select>
            </label>
          </div>
          <div
            class="w-full flex flex-col max-w-[102px] max-h-[36px]"
            data-slot="mainWrapper"
          >
            <button
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-label="appointment status"
              aria-labelledby="react-aria-:rh: react-aria-:rc:"
              class="relative gap-3 w-full inline-flex flex-row items-center shadow-sm tap-highlight-transparent group-data-[focus=true]:bg-default-100 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-2.5 py-2 h-[46px] rounded-medium bg-background-100 shadow-stack border-1 outline-offset-0 data-[focus=true]:border-green data-[focus=true]:border-1 data-[focus=true]:bg-background-100 data-[hover=true]:bg-background-100 group-data-[invalid=true]:!border-danger-100 min-h-[36px]"
              data-slot="trigger"
              id="react-aria-:rc:"
              type="button"
            >
              <div
                class="inline-flex h-full min-h-4 items-center gap-1.5 box-border text-xs leading-[18px] group-data-[invalid=true]:!text-danger-100 w-[80px]"
                data-slot="innerWrapper"
              >
                <span
                  class="w-full text-left rtl:text-right truncate group-data-[has-value=true]:text-default-foreground text-xs group-data-[invalid=true]:!text-danger-100 text-primary-100 font-medium"
                  data-slot="value"
                  id="react-aria-:rh:"
                >
                  New
                </span>
              </div>
              <svg
                aria-hidden="true"
                class="absolute end-3 w-4 h-4 transition-transform duration-150 ease motion-reduce:transition-none data-[open=true]:rotate-180 text-primary-100"
                data-slot="selectorIcon"
                fill="none"
                focusable="false"
                height="1em"
                role="presentation"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                width="1em"
              >
                <path
                  d="m6 9 6 6 6-6"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="flex flex-col relative gap-4 w-full"
    >
      <div
        class="p-4 z-0 flex flex-col relative justify-between gap-4 bg-content1 overflow-auto rounded-large shadow-small w-full mocked-class"
      >
        <table
          aria-describedby=""
          aria-label="Table"
          class="min-w-full h-auto table-auto w-full"
          id="appointment-upcoming"
          role="grid"
          tabindex="0"
        >
          <thead
            class="[&>tr]:first:rounded-lg hidden"
            role="rowgroup"
          >
            <tr
              class="group outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2"
              role="row"
            >
              <th
                class="group px-3 h-10 align-middle bg-default-100 whitespace-nowrap text-foreground-500 text-tiny font-semibold first:rounded-s-lg last:rounded-e-lg data-[sortable=true]:cursor-pointer data-[hover=true]:text-foreground-400 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-start"
                data-key="startTime"
                id="appointment-upcoming-startTime"
                role="columnheader"
                tabindex="-1"
              >
                Start time
              </th>
              <th
                class="group px-3 h-10 align-middle bg-default-100 whitespace-nowrap text-foreground-500 text-tiny font-semibold first:rounded-s-lg last:rounded-e-lg data-[sortable=true]:cursor-pointer data-[hover=true]:text-foreground-400 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-start"
                data-key="senderId"
                id="appointment-upcoming-senderId"
                role="columnheader"
                tabindex="-1"
              >
                Sender
              </th>
              <th
                class="group px-3 h-10 align-middle bg-default-100 whitespace-nowrap text-foreground-500 text-tiny font-semibold first:rounded-s-lg last:rounded-e-lg data-[sortable=true]:cursor-pointer data-[hover=true]:text-foreground-400 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-start"
                data-key="durationTime"
                id="appointment-upcoming-durationTime"
                role="columnheader"
                tabindex="-1"
              >
                Duration time
              </th>
              <th
                class="group px-3 h-10 align-middle bg-default-100 whitespace-nowrap text-foreground-500 text-tiny font-semibold first:rounded-s-lg last:rounded-e-lg data-[sortable=true]:cursor-pointer data-[hover=true]:text-foreground-400 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-start"
                data-key="actions"
                id="appointment-upcoming-actions"
                role="columnheader"
                tabindex="-1"
              >
                Actions
              </th>
            </tr>
            <tr
              aria-hidden="true"
              class="w-px h-px block"
              style="margin-left: 0.25rem; margin-top: 0.25rem;"
              tabindex="-1"
            />
          </thead>
          <tbody
            role="rowgroup"
          >
            <tr
              aria-labelledby="appointment-upcoming-1-startTime"
              class="group outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2"
              data-first="true"
              data-key="1"
              role="row"
              tabindex="-1"
            >
              <td
                class="py-2 px-3 relative align-middle whitespace-normal text-small font-normal [&>*]:z-1 [&>*]:relative outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]:text-foreground-300 group-data-[disabled=true]:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start mocked-class"
                data-key="table-row-cell-startTime"
                headers="appointment-upcoming-startTime"
                id="appointment-upcoming-1-startTime"
                role="cell"
                tabindex="-1"
              >
                <div
                  class="rounded-md w-[30px] md:w-[37px] h-10 bg-background-100 text-center pt-1"
                >
                  <p
                    class="mocked-class"
                  >
                    Fri
                  </p>
                  <p
                    class="mocked-class"
                  >
                    11
                  </p>
                </div>
              </td>
              <td
                class="py-2 px-3 relative align-middle whitespace-normal text-small font-normal [&>*]:z-1 [&>*]:relative outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]:text-foreground-300 group-data-[disabled=true]:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start mocked-class"
                data-key="table-row-cell-senderId"
                headers="appointment-upcoming-senderId"
                role="cell"
                tabindex="-1"
              >
                <p
                  class="mocked-class"
                >
                  Alex Stanton
                </p>
                <p
                  class="mocked-class"
                >
                  1:30 pm to 3:00 pm
                </p>
              </td>
              <td
                class="py-2 px-3 relative align-middle whitespace-normal text-small font-normal [&>*]:z-1 [&>*]:relative outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]:text-foreground-300 group-data-[disabled=true]:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start mocked-class"
                data-key="table-row-cell-durationTime"
                headers="appointment-upcoming-durationTime"
                role="cell"
                tabindex="-1"
              >
                <p
                  class="mocked-class"
                >
                  1:30 pm to 3:00 pm
                </p>
              </td>
              <td
                class="py-2 px-3 relative align-middle whitespace-normal text-small font-normal [&>*]:z-1 [&>*]:relative outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]:text-foreground-300 group-data-[disabled=true]:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start mocked-class"
                data-key="table-row-cell-actions"
                headers="appointment-upcoming-actions"
                role="cell"
                tabindex="-1"
              >
                <div
                  class="flex justify-end"
                >
                  <button
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-label="Dropdown button"
                    class="group relative items-center box-border appearance-none select-none whitespace-nowrap font-normal overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default text-default-foreground data-[hover=true]:opacity-hover z-10 aria-expanded:scale-[0.97] aria-expanded:opacity-70 subpixel-antialiased mocked-class flex justify-start capitalize"
                    data-slot="trigger"
                    id="react-aria-:rk:"
                    type="button"
                  >
                    <div
                      class="text-primary-100 max-h-6 max-w-6"
                    >
                      <svg
                        class="w-[11px] h-[11px] md:w-4 md:h-4"
                        fill="none"
                        viewBox="0 0 16 16"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M8 6C6.90076 6 6 6.90076 6 8C6 9.09924 6.90076 10 8 10C9.09924 10 10 9.09924 10 8C10 6.90076 9.09924 6 8 6Z"
                          fill="currentColor"
                        />
                        <path
                          d="M8 0C6.90076 0 6 0.900764 6 2C6 3.09924 6.90076 4 8 4C9.09924 4 10 3.09924 10 2C10 0.900764 9.09924 0 8 0Z"
                          fill="currentColor"
                        />
                        <path
                          d="M8 12C6.90076 12 6 12.9008 6 14C6 15.0992 6.90076 16 8 16C9.09924 16 10 15.0992 10 14C10 12.9008 9.09924 12 8 12Z"
                          fill="currentColor"
                        />
                      </svg>
                    </div>
                  </button>
                </div>
              </td>
            </tr>
            <tr
              aria-labelledby="appointment-upcoming-2-startTime"
              class="group outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2"
              data-key="2"
              data-middle="true"
              data-odd="true"
              role="row"
              tabindex="-1"
            >
              <td
                class="py-2 px-3 relative align-middle whitespace-normal text-small font-normal [&>*]:z-1 [&>*]:relative outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]:text-foreground-300 group-data-[disabled=true]:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start mocked-class"
                data-key="table-row-cell-startTime"
                headers="appointment-upcoming-startTime"
                id="appointment-upcoming-2-startTime"
                role="cell"
                tabindex="-1"
              >
                <div
                  class="rounded-md w-[30px] md:w-[37px] h-10 bg-background-100 text-center pt-1"
                >
                  <p
                    class="mocked-class"
                  >
                    Sat
                  </p>
                  <p
                    class="mocked-class"
                  >
                    12
                  </p>
                </div>
              </td>
              <td
                class="py-2 px-3 relative align-middle whitespace-normal text-small font-normal [&>*]:z-1 [&>*]:relative outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]:text-foreground-300 group-data-[disabled=true]:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start mocked-class"
                data-key="table-row-cell-senderId"
                headers="appointment-upcoming-senderId"
                role="cell"
                tabindex="-1"
              >
                <p
                  class="mocked-class"
                >
                  Alex Stanton
                </p>
                <p
                  class="mocked-class"
                >
                  9:00 am to 10:00 am
                </p>
              </td>
              <td
                class="py-2 px-3 relative align-middle whitespace-normal text-small font-normal [&>*]:z-1 [&>*]:relative outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]:text-foreground-300 group-data-[disabled=true]:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start mocked-class"
                data-key="table-row-cell-durationTime"
                headers="appointment-upcoming-durationTime"
                role="cell"
                tabindex="-1"
              >
                <p
                  class="mocked-class"
                >
                  9:00 am to 10:00 am
                </p>
              </td>
              <td
                class="py-2 px-3 relative align-middle whitespace-normal text-small font-normal [&>*]:z-1 [&>*]:relative outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]:text-foreground-300 group-data-[disabled=true]:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start mocked-class"
                data-key="table-row-cell-actions"
                headers="appointment-upcoming-actions"
                role="cell"
                tabindex="-1"
              >
                <div
                  class="flex justify-end"
                >
                  <button
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-label="Dropdown button"
                    class="group relative items-center box-border appearance-none select-none whitespace-nowrap font-normal overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default text-default-foreground data-[hover=true]:opacity-hover z-10 aria-expanded:scale-[0.97] aria-expanded:opacity-70 subpixel-antialiased mocked-class flex justify-start capitalize"
                    data-slot="trigger"
                    id="react-aria-:rn:"
                    type="button"
                  >
                    <div
                      class="text-primary-100 max-h-6 max-w-6"
                    >
                      <svg
                        class="w-[11px] h-[11px] md:w-4 md:h-4"
                        fill="none"
                        viewBox="0 0 16 16"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M8 6C6.90076 6 6 6.90076 6 8C6 9.09924 6.90076 10 8 10C9.09924 10 10 9.09924 10 8C10 6.90076 9.09924 6 8 6Z"
                          fill="currentColor"
                        />
                        <path
                          d="M8 0C6.90076 0 6 0.900764 6 2C6 3.09924 6.90076 4 8 4C9.09924 4 10 3.09924 10 2C10 0.900764 9.09924 0 8 0Z"
                          fill="currentColor"
                        />
                        <path
                          d="M8 12C6.90076 12 6 12.9008 6 14C6 15.0992 6.90076 16 8 16C9.09924 16 10 15.0992 10 14C10 12.9008 9.09924 12 8 12Z"
                          fill="currentColor"
                        />
                      </svg>
                    </div>
                  </button>
                </div>
              </td>
            </tr>
            <tr
              aria-labelledby="appointment-upcoming-3-startTime"
              class="group outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2"
              data-key="3"
              data-last="true"
              role="row"
              tabindex="-1"
            >
              <td
                class="py-2 px-3 relative align-middle whitespace-normal text-small font-normal [&>*]:z-1 [&>*]:relative outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]:text-foreground-300 group-data-[disabled=true]:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start mocked-class"
                data-key="table-row-cell-startTime"
                headers="appointment-upcoming-startTime"
                id="appointment-upcoming-3-startTime"
                role="cell"
                tabindex="-1"
              >
                <div
                  class="rounded-md w-[30px] md:w-[37px] h-10 bg-background-100 text-center pt-1"
                >
                  <p
                    class="mocked-class"
                  >
                    Fri
                  </p>
                  <p
                    class="mocked-class"
                  >
                    13
                  </p>
                </div>
              </td>
              <td
                class="py-2 px-3 relative align-middle whitespace-normal text-small font-normal [&>*]:z-1 [&>*]:relative outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]:text-foreground-300 group-data-[disabled=true]:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start mocked-class"
                data-key="table-row-cell-senderId"
                headers="appointment-upcoming-senderId"
                role="cell"
                tabindex="-1"
              >
                <p
                  class="mocked-class"
                >
                  ThangHq
                </p>
                <p
                  class="mocked-class"
                >
                  10:00 am to 11:00 am
                </p>
              </td>
              <td
                class="py-2 px-3 relative align-middle whitespace-normal text-small font-normal [&>*]:z-1 [&>*]:relative outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]:text-foreground-300 group-data-[disabled=true]:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start mocked-class"
                data-key="table-row-cell-durationTime"
                headers="appointment-upcoming-durationTime"
                role="cell"
                tabindex="-1"
              >
                <p
                  class="mocked-class"
                >
                  10:00 am to 11:00 am
                </p>
              </td>
              <td
                class="py-2 px-3 relative align-middle whitespace-normal text-small font-normal [&>*]:z-1 [&>*]:relative outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]:text-foreground-300 group-data-[disabled=true]:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start mocked-class"
                data-key="table-row-cell-actions"
                headers="appointment-upcoming-actions"
                role="cell"
                tabindex="-1"
              >
                <div
                  class="flex justify-end"
                >
                  <button
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-label="Dropdown button"
                    class="group relative items-center box-border appearance-none select-none whitespace-nowrap font-normal overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-default text-default-foreground data-[hover=true]:opacity-hover z-10 aria-expanded:scale-[0.97] aria-expanded:opacity-70 subpixel-antialiased mocked-class flex justify-start capitalize"
                    data-slot="trigger"
                    id="react-aria-:rq:"
                    type="button"
                  >
                    <div
                      class="text-primary-100 max-h-6 max-w-6"
                    >
                      <svg
                        class="w-[11px] h-[11px] md:w-4 md:h-4"
                        fill="none"
                        viewBox="0 0 16 16"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M8 6C6.90076 6 6 6.90076 6 8C6 9.09924 6.90076 10 8 10C9.09924 10 10 9.09924 10 8C10 6.90076 9.09924 6 8 6Z"
                          fill="currentColor"
                        />
                        <path
                          d="M8 0C6.90076 0 6 0.900764 6 2C6 3.09924 6.90076 4 8 4C9.09924 4 10 3.09924 10 2C10 0.900764 9.09924 0 8 0Z"
                          fill="currentColor"
                        />
                        <path
                          d="M8 12C6.90076 12 6 12.9008 6 14C6 15.0992 6.90076 16 8 16C9.09924 16 10 15.0992 10 14C10 12.9008 9.09924 12 8 12Z"
                          fill="currentColor"
                        />
                      </svg>
                    </div>
                  </button>
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
